operate/importer-8_4/src/main/java/io/camunda/operate/zeebeimport/v8_4/processors/FlowNodeInstanceZeebeRecordProcessor.java

```diff
@@ -64,7 +69,21 @@ public FlowNodeInstanceZeebeRecordProcessor(
       final PartitionHolder partitionHolder,
       final Metrics metrics) {
     this.flowNodeInstanceTemplate = flowNodeInstanceTemplate;
-    final var flowNodeTreeCacheSize = operateProperties.getImporter().getFlowNodeTreeCacheSize();
+    flowNodeTreeCacheSize = operateProperties.getImporter().getFlowNodeTreeCacheSize();
+    this.partitionHolder = partitionHolder;
+    this.flowNodeStore = flowNodeStore;
+    this.metrics = metrics;
+  }
+
+  private FNITransformer lazyLoadFNITransformer() {
+    if (fniTransformer != null) {
+      return fniTransformer;
+    }
+
+    // We create the FNITransformer lazy, as partition holder didn't hold the partitionIds right on
+    // start. The chances are higher that when we start importing we can also request the
+    // partitions.
```

Comment:
I think that's not only more likely but a given. When scheduling the records readers (via `io.camunda.operate.zeebeimport.RecordsReaderHolder.getAllRecordsReaders()` and calling code), then we already create records readers per partition. Not batch makes it into processing before that.